lib/shape.ml

```diff
@@ -60,27 +62,34 @@ let line ?x1 ?y1 x2 y2 =
   | Some x, Some y -> Line { a = { x; y }; b = { x = x2; y = y2 } }
   | _ -> Line { a = { x = 0; y = 0 }; b = { x = x2; y = y2 } }
 
-let translate dx dy shape =
+let complex arg = 
+  match arg with 
+  | [single] -> Complex [single] 
```

Comment:
We don't need to handle the singleton case separately.

`[a]` is actually `a :: nil`, so it is picked up by the second case.

Can you also rename `arg` to something more descriptive? (e.g. `shapes`) ?

I'm not sure if we should be in-validating the case with empty list. It would be convenient to have when e.g. using `repeat` with  n = 0. Can you accept it for now and then we will discuss whether it should be invalidated or not?